cyclonedx/model/composition.py

```diff
@@ -0,0 +1,231 @@
+from sortedcontainers import SortedSet
+
+from .._internal.compare import ComparableTuple as _ComparableTuple
+from ..serialization import BomRefHelper
+from .bom_ref import BomRef
+
+
+@serializable.serializable_enum
+class AggregateType(str, Enum):
+    """
+    This is our internal representation of the composition.aggregate ENUM type within the CycloneDX standard.
+
+    .. note::
+        Introduced in CycloneDX v1.4
+
+    .. note::
+        See the CycloneDX Schema for hashType: https://cyclonedx.org/docs/1.4/xml/#type_aggregateType
+    """
+
+    """
+    The relationship is complete. No further relationships including constituent components, services, or dependencies
+    are known to exist.
+    """
+    COMPLETE = 'complete'
+
+    """
+    The relationship is incomplete. Additional relationships exist and may include constituent components, services, or
+    dependencies.
+    """
+    INCOMPLETE = 'incomplete'
+
+    """
+    The relationship is incomplete. Only relationships for first-party components, services, or their dependencies are
+    represented.
+    """
+    INCOMPLETE_FIRST_PARTY_ONLY = 'incomplete_first_party_only'
+
+    """
+    The relationship is incomplete. Only relationships for first-party components, services, or their dependencies are
+    represented, limited specifically to those that are proprietary.
+    """
+    INCOMPLETE_FIRST_PARTY_PROPRIETARY_ONLY = 'incomplete_first_party_proprietary_only'
+
+    """
+    The relationship is incomplete. Only relationships for first-party components, services, or their dependencies are
+    represented, limited specifically to those that are opensource.
+    """
+    INCOMPLETE_FIRST_PARTY_OPENSOURCE_ONLY = 'incomplete_first_party_opensource_only'
+
+    """
+    The relationship is incomplete. Only relationships for third-party components, services, or their dependencies are
+    represented.
+    """
+    INCOMPLETE_THIRD_PARTY_ONLY = 'incomplete_third_party_only'
+
+    """
+    The relationship is incomplete. Only relationships for third-party components, services, or their dependencies are
+    represented, limited specifically to those that are proprietary.
+    """
+    INCOMPLETE_THIRD_PARTY_PROPRIETARY_ONLY = 'incomplete_third_party_proprietary_only'
+
+    """
+    The relationship is incomplete. Only relationships for third-party components, services, or their dependencies are
+    represented, limited specifically to those that are opensource.
+    """
+    INCOMPLETE_THIRD_PARTY_OPENSOURCE_ONLY = 'incomplete_third_party_opensource_only'
+
+    """
+    The relationship may be complete or incomplete. This usually signifies a 'best-effort' to obtain constituent
+    components, services, or dependencies but the completeness is inconclusive.
+    """
+    UNKNOWN = 'unknown'
+
+    """
+    The relationship completeness is not specified.
+    """
+    NOT_SPECIFIED = 'not_specified'
+
+
+@serializable.serializable_class
+class CompositionReference:
```

Comment:
what is the purpose of this class?
why not use simple `BomRef` instances instead?

Reply:
Yes - it was added for structural reasons - happy to leave as is @jkowalleck ?

Reply:
i do not understand. what were these structural reasons? 
I mean everybody ysing the library would ask the same question I did.


from the schema it looks like all these `compositions.assembies` and `compositions.dependencies` are simple sets of `BomRef`. 
see https://github.com/CycloneDX/specification/blob/8e131b1688ccfe41e1bfdd4b3280f33dcc06d04c/schema/bom-1.6.schema.json#L2235-L2252